src/scene/shader-lib/chunks/standard/frag/frontend.js

```diff
@@ -0,0 +1,37 @@
+export default /* glsl */`
+struct Frontend {
+    vec2 uvOffset;
+
+    float alpha;
+
+    vec3 worldNormal;
+
+    vec3 albedo;
+
+    float transmission;
+    float thickness;
+
+    vec3 iridescenceFresnel;
+    float iridescence;
+    float iridescenceThickness;
+
+    vec3 specularity;
+    float glossiness;
+    float metalness;
+    float specularityFactor;
+
+    float ao;
+
+    vec3 emission;
+
+    float clearcoatSpecularity;
+    float clearcoatGlossiness;
+    vec3 clearcoatWorldNormal;
+
+    float sheenGlossiness;
+    vec3 sheenSpecularity;
+
+    vec3 lightmap;
+    vec3 lightmapDir;
```

Comment:
I wonder if we should only have `lightmap' here, as lightmapDir is only related to our directional lightmap implementation. Perhaps by exposing ligthmapDir we lock frontend to just this specific basis? Maybe the front end needs to evaluate ligthmap with dir, and give backend just the result?